File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    rank = {}
    for i in range(len(players)):
        rank[players[i]] = i
        
    for call in callings:
        now = rank[call]
        rank[call] -= 1
        rank[players[now - 1]] += 1
        players[now], players[now - 1] = players[now - 1], players[now]
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    var answer = [];
    const playerList = players.reduce((acc, cur, idx) => {
        acc[cur] = idx
        return acc
    }, {})
    
    callings.forEach(call => {
        const curIdx = playerList[call]
        const nextIdx = curIdx - 1
        const nextPlayer = players[nextIdx]
        players[curIdx] = nextPlayer
        players[nextIdx] = call
        playerList[call]--
        playerList[nextPlayer]++
    })
    
    return players;
}